### PR TITLE
feat(mail): unknown-flag fuzzy-match for lark-cli mail domain

### PR DESCRIPTION
**Pull request by xzcong0820**
Generated by the harness-coding skill.

- **Task ID**: 01KR5EDZE944JYRXCN6M99JCD1-9
- **Branch**: harness/01kr5edze944jyrxcn6m99jcd1
- **Target**: main

## Summary

Adds mail-domain unknown-flag fuzzy-match → JSON ErrorEnvelope with did-you-mean candidates for `lark-cli mail`. Strictly scoped to `shortcuts/mail/` + a 1-line branch in `shortcuts/register.go`; non-mail domains are untouched. Exit code is unchanged (still `1` via `ExitAPI`, not `ExitValidation`) — no breaking change to callers parsing exit codes.

3 files changed, 508 insertions:
- `shortcuts/mail/flag_suggest.go` — implementation (`InstallOnMail`, `parseUnknownToken`, `collectFlags`, `suggest` w/ prefix + Levenshtein DP, `suggestShorthand`, `buildHint`).
- `shortcuts/mail/flag_suggest_test.go` — 12 table-driven tests covering prefix / Levenshtein / hidden-skipped / top-N stable sort / shorthand exact + prefix fallback / parse regex / errorFunc pass-through / exit-code regression.
- `shortcuts/register.go` — +3 lines: `if service == "mail" { mail.InstallOnMail(svc) }` after the mount loop.

## Sprints

| ID | Title | Status | Commit |
|----|-------|--------|--------|
| S1 | Synthesize transport contract for larksuite/cli (mail flag_suggest) | passed | — |
| S2 | Implement mail-domain unknown-flag fuzzy match + ErrorEnvelope | passed | 5ae9c2d |

## Source specs

- [V5 tech-design](https://bytedance.larkoffice.com/docx/CQuEdU5xaoPbFBx3EeecWFMLnHb)
- /tmp/harness-agent/coding-dev/01KR5EDZE944JYRXCN6M99JCD1/01KR5EDZE944JYRXCN6M99JCD1-9/input/tech-design.md

---
This MR was created autonomously. Quality gates were enforced by the repo's own pre-commit hooks.

## Summary by CodeRabbit

* **New Features**
  * Mail commands now provide intelligent unknown-flag suggestions (up to 5 candidates) with concise “Did you mean … ?” hints or a help hint.

* **Behavior**
  * Registering the Mail service installs a mail-specific flag-suggestion hook so unknown-flag errors return a structured, user-facing error and consistent exit code.

* **Tests**
  * Added extensive unit and integration tests for suggestion ranking, shorthand handling, hidden-flag exclusion, parsing edge cases, and exit-error formatting.

[Review Change Stack](https://app.coderabbit.ai/change-stack/larksuite/cli/pull/806)

### CODE REVIEW

File: shortcuts/mail/flag_suggest.go [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+func min3(a, b, c int) int {
+	m := a
+	if b < m {
+		m = b
+	}
+	if c < m {
+		m = c
+	}
+	return m
+}
```
Comment (chanthuang) [inline shortcuts/mail/flag_suggest.go]: **Issue:** Go 1.23 (per `go.mod`) supports the built-in variadic `min()`. The custom `min3` helper can be replaced with `min(a, b, c)` — one fewer function to maintain.

**Severity:** 💡 Suggestion

**Suggested Fix:**
```go
curr[j] = min(curr[j-1]+1, prev[j]+1, prev[j-1]+cost)
```
Then remove `min3`.

File: shortcuts/mail/flag_suggest.go [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+// flagSuggestErrorFunc converts pflag's unknown-flag errors into a
+// structured *output.ExitError carrying candidate suggestions. Any other
+// error is passed through unchanged so cobra's existing handling kicks in.
+func flagSuggestErrorFunc(c *cobra.Command, err error) error {
+	if err == nil {
+		return nil
+	}
+	token, isShorthand, ok := parseUnknownToken(err.Error())
```
Comment (chanthuang) [inline shortcuts/mail/flag_suggest.go:66]: **Issue:** `flagSuggestErrorFunc` is set on the mail parent `svc` and cobra walks up the parent chain for the nearest non-nil hook — this works correctly. However, there's no integration-level test that exercises the full path: `RegisterShortcuts` → `InstallOnMail` → unknown flag on a real mail subcommand → structured `ExitError`. The 12 unit tests cover the internal logic well, but a single end-to-end test in `register_test.go` would guard against the wiring breaking silently (e.g., if the `if service == "mail"` branch is accidentally removed or the package import changes).

**Severity:** 💡 Suggestion

**Suggested Fix:** Add a test similar to `TestRegisterShortcutsMountsBaseCommands` that calls `RegisterShortcuts`, looks up a mail subcommand, sets `FlagErrorFunc` to fire via an invalid flag, and asserts the returned error is an `*output.ExitError` with `Type: "unknown_flag"`.

File: shortcuts/mail/flag_suggest.go (hunk 1 above)
Comment (xzcong0820) [reply shortcuts/mail/flag_suggest.go]: 已按建议改用 Go 1.23 内置 `min`，并删除 `min3` helper。见 35745b5。

File: shortcuts/mail/flag_suggest.go (hunk 2 above)
Comment (xzcong0820) [reply shortcuts/mail/flag_suggest.go:66]: 已在 `shortcuts/register_test.go` 加两条集成测试：

- `TestRegisterShortcutsInstallsMailFlagSuggestHook`：验证 mail 父命令装上了 `*output.ExitError{Type:"unknown_flag"}` 的结构化错误 hook（防止 `if service == "mail"` 分支被静默改坏）。
- `TestRegisterShortcutsLeavesNonMailFlagErrorUntouched`：验证 hook 严格限定在 mail 域，不会泄漏到其它 service。

见 35745b5。

File: shortcuts/register_test.go [hunk 3]
```diff
@@ -305,6 +307,62 @@ func TestRegisterShortcutsReusesExistingServiceCommand(t *testing.T) {
...
+		t.Fatalf("find base command: %v", err)
+	}
+	in := errors.New("unknown flag: --bogus")
+	got := baseCmd.FlagErrorFunc()(baseCmd, in)
+	// Default cobra hook is identity — anything else means the mail hook
+	// leaked across domains.
+	var exitErr *output.ExitError
+	if errors.As(got, &exitErr) {
+		t.Fatalf("base service unexpectedly produced *output.ExitError: %#v", exitErr)
+	}
```
[the remark this answers is not in the document]
Comment (xzcong0820) [reply shortcuts/register_test.go:363]: Applied — added `got == in` pointer-equality assertion to lock the Cobra identity contract on non-mail FlagErrorFunc. See 35745b5.

File: shortcuts/mail/flag_suggest_test.go [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+		{long: "ccc"},
+		{long: "ddd"},
+		{long: "eee"},
+		{long: "fff"},
+	}
+	got := suggest("zzz", names)
+	require.LessOrEqual(t, len(got), maxCandidates)
+	// At equal distances results must be sorted by Flag asc.
+	for i := 1; i < len(got); i++ {
+		assert.LessOrEqual(t, got[i-1].Flag, got[i].Flag, "candidates must be sorted by Flag asc on ties")
```
[the remark this answers is not in the document]
Comment (xzcong0820) [reply shortcuts/mail/flag_suggest_test.go]: Good catch — the original test passed trivially because `levThreshold("zzz")=2` but every candidate was at distance 3, so `suggest` returned an empty slice and the cap/tiebreak assertions were vacuous.

Switched to `unknown="aaaa"` + names `aaab..aaag` (all distance 1, within threshold). Now actually verifies `len(got)==maxCandidates` and the top-5 are sorted `--aaab..--aaaf` by Flag asc. See 4dbbe72.

File: shortcuts/mail/flag_suggest.go [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+		matches = suggestShorthand(token, names)
+	} else {
+		matches = suggest(token, names)
+	}
+	hint := buildHint(c, matches)
+	detail := map[string]any{
+		"unknown":      rawUnknownToken(token, isShorthand),
+		"command_path": c.CommandPath(),
+		"candidates":   matches,
+	}
```
[the remark this answers is not in the document]
Comment (xzcong0820) [reply shortcuts/mail/flag_suggest.go:90]: Applied — added `if matches == nil { matches = []Candidate{} }` before building the detail map, plus a new test `TestFlagSuggestErrorFunc_CandidatesAlwaysArray` (uses a cobra command with zero flags to force the nil path) that locks the JSON wire shape to `candidates: []` instead of `null`. See 4dbbe72.

**Review by chanthuang — APPROVED** [review]
